src/MongoDB/BulkWrite.c

```diff
@@ -45,6 +45,32 @@ static inline bool php_phongo_bulkwrite_insert_is_legacy_index(bson_t *bdocument
 	return false;
 } /* }}} */
 
+/* Extracts the "_id" field of a BSON document into a return value. */
+static void php_phongo_bulkwrite_extract_id(bson_t *doc, zval **return_value) /* {{{ */
+{
+	php_phongo_bson_state state = PHONGO_BSON_STATE_INITIALIZER;
+	zval *id = NULL;
+
+	state.map.root_type = PHONGO_TYPEMAP_NATIVE_ARRAY;
+
+	if (!php_phongo_bson_to_zval_ex(bson_get_data(doc), doc->len, &state)) {
+		goto cleanup;
+	}
+
+#if PHP_VERSION_ID >= 70000
+	id = php_array_fetchc(&state.zchild, "_id");
+#else
+	id = php_array_fetchc(state.zchild, "_id");
+#endif
+
+	if (id) {
+		ZVAL_ZVAL(*return_value, id, 1, 0);
```

Comment:
@derickr: I verified that we don't leak/crash if this were to hypothetically fail to find the "_id" field. In that case, `return_value` remains `NULL`.